tests/test_appclass.py

```diff
@@ -35,3 +39,42 @@ def test_dependencies(generate_app):
     with pytest.raises(traitlets.TraitError):
         app.version_to_install = "v22.11.0"
     app.version_to_install = "v22.11.1"
+
+
+def test_app_watch(tmp_path):
+    """Test the aiidalab app watch responsive to the app path changes."""
+
+    @dataclass
+    class DummyApp:
+        path: Path
+        x: int = 0
+
+        def refresh_async(self):
+            self.x += 1
+
+    app = DummyApp(path=Path(tmp_path))
+    app_watch = AiidaLabAppWatch(app)
+    app_watch.start()
+
+    # The observer is start in a thread so need to wait until it is alive
+    while app_watch._observer is None or not app_watch._observer.is_alive():
+        sleep(0.1)
+
+    # Trigger action by file events
+    # touch a file will trigger action 4 times: create and close file and two modifies of folder
+    testfile = tmp_path / "test0"
+    testfile.touch()
+
+    app_watch.stop()
+    app_watch.join()
```

Comment:
Thanks. I think it is safer to use a timeout and then assert that the two thread stopped, otherwise we risk a potential deadlock in our CI. We can also check the number of threads before and after. 

```suggestion
    assert threading.active_count() > 1

    app_watch.stop()
    app_watch.join(timeout=10.)
    assert threading.active_count() == 1
```

Reply:
Nice advice, more robust for the test involved with threading. changed.

Reply:
Did you push the change? I don't see it on Github.

Reply:
Gosh, what a failure. I push now. 😢